airbyte-cdk/python/unit_tests/test_connector.py

```diff
@@ -75,6 +83,11 @@ def test_read_config(nonempty_file, integration: Connector, mock_config):
     assert mock_config == actual
 
 
+def test_read_non_json_config(nonjson_file, integration: Connector):
```

Comment:
+1 to the test using the invalid config. But can we also add a test when we run `read_state()` where the incoming state comes in the form of a list to make sure the underlying `_read_json_file()` handles it correctly as expected

Reply:
Is it what you were looking for @brianjlai ? https://github.com/airbytehq/airbyte/pull/20019/commits/67c55b8b1d2b0b6e0e16fae4f9f54dd224e83c19

Reply:
ah yeah okay sounds like we already got tests. so don't mind me then